src/Schema.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace BuildEmpire\Apartment;
+
+use BuildEmpire\Apartment\Exceptions\NoSchemaSetException;
+use BuildEmpire\Apartment\Exceptions\NoSchemaFoundException;
+use BuildEmpire\Apartment\Exceptions\SchemaNameNotValidException;
+use BuildEmpire\Apartment\Helpers\ApartmentHelpers;
+
+class Schema
+{
+    protected $schemaName = false;
+
+    /**
+     * Set the schema name.
+     *
+     * @param $schemaName
+     * @throws SchemaNameNotValidException
+     */
+    public function setSchemaName($schemaName)
+    {
+        if (!ApartmentHelpers::isSchemaNameValid($schemaName)) {
+            throw new SchemaNameNotValidException('The apartment ' . $schemaName . ' is not valid. It must be all lowercase and only contain letters, numbers, or underscores.');
+        }
+        $this->schemaName = $schemaName;
+    }
+
+    /**
+     * Try to get the schema name.
+     *
+     * @return string
+     * @throws NoSchemaSetException
+     */
+    public function getSchemaName()
+    {
+        if ($this->schemaName !== false) {
+            return $this->schemaName;
+        }
+
+        throw new NoSchemaSetException('No schema is set. If you want to check if a schema exists use doesSchemaExist method.');
```

Comment:
I'm a little uncomfortable with a `get` which throws, is this conventional/consistent in the php/laravel/etc. world?